diffkemp/diffkemp.py

```diff
@@ -237,8 +238,17 @@ def compare(args):
         for fun, old_fun_desc in sorted(group.functions.items()):
             # Check if the function exists in the other snapshot
             new_fun_desc = new_functions.get_by_name(fun, group_name)
-            if not (new_fun_desc and
-                    old_fun_desc.mod.has_function(fun) and
+            if not new_fun_desc:
+                continue
+
+            # Check if the module exists in both snapshots
+            if old_fun_desc.mod is None or new_fun_desc.mod is None:
+                result.add_inner(Result(Result.Kind.UNKNOWN, fun, fun))
+                print("{}: unknown".format(fun))
+                continue
+
+            # Check if the function exists in both modules
+            if not (old_fun_desc.mod.has_function(fun) and
```

Comment:
I suggest we could move this check into the `generate` phase, i.e., do not include symbols that are not functions into `functions.yaml`. I believe that currently, we include global variables and as the LLVM file we give the one where the variable is defined, which is useless. Also, this would be more efficient (it is expected that `compare` is run more often than `generate`).

Reply:
That's a good idea, the LLVM IR parsing also takes unnecessary time.

Reply:
Well, `has_function` only searches for regex, but it still takes some time. 
Can you include this into this PR?

Reply:
I've included both this and the better wording.